### PR TITLE
EDM-281: API returns aggregate device status for fleets

### PR DESCRIPTION
**Pull request by avishayt**
deviceSummary:
  bySummaryStatus:
    an excellent summary: 1000
    decent summary: 1000
    orange: 1000
    terrible summary: 1000
    tractor: 1000
  byUpdateStatus:
    all good in the hood: 1000
    motorcycle: 1000
    somewhat: 1000
    strawberry: 1000
    yech: 1000
  total: 5000

### CODE REVIEW

File: api/v1alpha1/openapi.yaml [hunk 1]
```diff
@@ -3433,9 +3439,32 @@ components:
           description: 'Current state of the fleet.'
           items:
             $ref: '#/components/schemas/Condition'
+        deviceSummary:
```
Comment (hexfusion) [inline api/v1alpha1/openapi.yaml:3442]: is this `devicesSummary`?

Comment (avishayt) [reply api/v1alpha1/openapi.yaml]: Sure, changed

File: api/v1alpha1/openapi.yaml [hunk 2]
```diff
@@ -3433,9 +3439,32 @@ components:
...
+      description: A summary of the devices in the fleet returned when fetching a single Fleet.
+      required:
+      - total
+      - bySummaryStatus
+      - byUpdateStatus
+      properties:
+        total:
+          type: integer
+          description: The total number of devices in the fleet.
+        bySummaryStatus:
```
[the remark this answers is not in the document]
Comment (avishayt) [reply api/v1alpha1/openapi.yaml]: OK by me

File: internal/store/fleet.go [hunk 3]
```diff
@@ -23,7 +23,7 @@ import (
 type Fleet interface {
 	Create(ctx context.Context, orgId uuid.UUID, fleet *api.Fleet, callback FleetStoreCallback) (*api.Fleet, error)
 	List(ctx context.Context, orgId uuid.UUID, listParams ListParams) (*api.FleetList, error)
-	Get(ctx context.Context, orgId uuid.UUID, name string) (*api.Fleet, error)
+	Get(ctx context.Context, orgId uuid.UUID, name string, withSummary bool) (*api.Fleet, error)
```
[the remark this answers is not in the document]
Comment (avishayt) [reply internal/store/fleet.go]: Love this pattern!  I changed it a little by adding a parameter, let me know.

File: internal/store/model/fleet.go [hunk 4]
```diff
@@ -69,7 +69,7 @@ func NewFleetFromApiResource(resource *api.Fleet) (*Fleet, error) {
 	}, nil
 }
 
-func (f *Fleet) ToApiResource() api.Fleet {
+func (f *Fleet) ToApiResource(summary *api.DeviceSummary) api.Fleet {
```
[the remark this answers is not in the document]
Comment (avishayt) [reply internal/store/model/fleet.go]: Done

**Review by avishayt — COMMENTED** [review]
Fixed, thanks for the review and great pattern!

**Review by hexfusion — APPROVED** [review]

**Pull request merged**